src/briefcase/integrations/android_sdk.py

```diff
@@ -982,23 +1054,36 @@ def update_emulator_config(self, avd, updates):
             for key, value in avd_config.items():
                 f.write(f"{key}={value}\n")
 
-    def start_emulator(self, avd):
+    def start_emulator(self, avd, test_mode=False):
         """Start an existing Android emulator.
 
         Returns when the emulator is booted and ready to accept apps.
 
         :param avd: The AVD of the device.
+        :param test_mode: Should the emulator run in test mode? (Default: False)
         """
         if avd not in set(self.emulators()):
             raise InvalidDeviceError("emulator AVD", avd)
 
+        if test_mode:
+            extra_args = [
+                "-no-window",
+                "-no-snapshot",
+                "-no-audio",
+                "-no-boot-anim",
+                "-wipe-data",
```

Comment:
There's slightly different reasoning for each argument:

* `-no-window` is because the window isn't (AFAICT) needed for the test to work, and it makes the startup slightly faster, which is beneficial for test conditions. If it turns out GUI tests need a window to work successfully, this one can be walked back. It's also needed for CI, AFAICT.
* `-no-snapshot` is another speed thing, with a side of ensuring that previous device state doesn't impact on test operation.
* `-no-audio` is mostly a CI thing; but also a "don't want my laptop beeping while it runs tests" thing. However, as with `-no-window`, if it's necessary to walk this back to make audio tests work, then I guess we'll need to do that.
* `-no-boot-anim is a test speed thing; it's actually [recommended by the Android docs](https://developer.android.com/studio/run/emulator-commandline).
* `-wipe-data` has two purposes. Firstly, it ensures test conditions are consistent, and that previous test runs won't impact on this one. However, the more pressing need is because `kill()` really is a kill - it can leave the device storage in a messy state, and wiping the data on startup ensures the emulator starts up cleanly. 

`kill()` is being invoked because if you're running headless, there's no visible process to remove the emulator, so you need to spelunk into `adb` to remove the emulator. AFAICT, there's no "clean" alternative to kill (i.e., a `shutdown`, rather than kill); all the documentation I can find says "just click the close button".

None of these are *required* for testing; and I believe `-no-window` are the only ones required for CI;  `-no-audio` *may* also be required for CI - I had some initial issues launching emulators, but I don't recall if I ran a test of "headless, but with audio".

I hadn't thought of the `-wipe-data` implications, though; that's a good catch.

At the very least, I agree it's desirable to expose a way to pass these options in at the command line, and then be more conservative with the options used in Briefcase by default.